srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRuleController.java

```diff
@@ -0,0 +1,163 @@
+import gov.va.med.srcalc.web.view.admin.EditRuleFactory;
+import gov.va.med.srcalc.web.view.admin.ValueMatcherBuilder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+/**
+ * Web MVC controller for editing existing rules.
+ */
+@Controller
+@RequestMapping("/admin/rules/{ruleId}")
+public class EditRuleController
+{
+    private static final Logger fLogger = LoggerFactory.getLogger(EditRuleController.class);
+    
+    private final AdminService fAdminService;
+    
+    /**
+     * Constructs an instance.
+     * @param adminService the AdminService to use for operations
+     * @throws NullPointerException if adminService is null
+     */
+    @Inject
+    public EditRuleController(final AdminService adminService)
+    {
+        fAdminService = Objects.requireNonNull(adminService);
+    }
+    
+    /**
+     * Creates an {@link EditExistingRule} instance for the identified
+     * rule.
+     * @param ruleId the id of the rule to edit
+     * @return the EditExistingRule instance
+     * @throws InvalidIdentifierException if no such rule exists
+     */
+    @ModelAttribute(NewRuleController.ATTRIBUTE_RULE)
+    private EditExistingRule createEditRuleFromRule(
+            @PathVariable("ruleId") final int ruleId) throws InvalidIdentifierException
+    {
+        fLogger.trace("Creating EditRule from previous Rule with ID: {}.", ruleId);
+        final Rule rule = fAdminService.getRuleById(ruleId);
+        return EditRuleFactory.getInstance(rule, fAdminService);
+    }
+    
+    @RequestMapping(method = RequestMethod.GET)
+    public ModelAndView displayEditRuleForm(
+            @ModelAttribute(NewRuleController.ATTRIBUTE_RULE) final EditExistingRule editRule,
+            @PathVariable("ruleId") final int ruleId)
+                    throws InvalidIdentifierException
+    {
+        final ModelAndView mav = new ModelAndView(Views.EDIT_RULE);
+        mav.addObject("SAVE_URL", ruleId);
+        mav.addObject("isNewRule", false);
+        NewRuleController.addVariablesToModel(mav, editRule, fAdminService);
+        return mav;
+    }
+    
+    @RequestMapping(method = RequestMethod.POST)
+    public ModelAndView handlePost(
+            @ModelAttribute(NewRuleController.ATTRIBUTE_RULE) final EditExistingRule editRule,
+            final BindingResult bindingResult,
+            @PathVariable("ruleId") final int ruleId,
+            @RequestParam(value = "submitButton") final String submitString) throws InvalidIdentifierException
```

Comment:
It is actually very easy to have Spring switch request handler methods based on this value. I am doing this for the model term upload. Check out [these docs](http://docs.spring.io/spring/docs/4.0.x/spring-framework-reference/html/mvc.html#mvc-ann-requestmapping-params-and-headers).